src/graph.py

```diff
@@ -1,82 +1,109 @@
-#         for node in list of nodes
-#             look for the node in the list of edges for the remaining nodes
-#             remove edge from the node
-
-# del_edge(n1, n2) deletes the edge connecting 'n1' and 'n2' from the graph, raises an eror if no such edge exists.
-#     if not in the list of edges raise error
-#     if in list
-#         delete edge from the n1
-
-# neighbors(n) returns the list of all nodes connected to 'n' by edges, raise an error if n is not in g.
-#     check has.node 
-#     if False, return error
-#     else return edges from node n (check with instructor regarding one way or two way arrows)
-
-# adjacent(n1,n2) returns true if there is an edge connecting n1 and n2, False if not, raises an error if either node not in list.
-#     check if nodes in list
-#     if false, return error
-#     if true check if n2 is an edge of n1
-#         if true return true,
-#         if not return false.
-
-
-
-
-
-
-
-
-
+        """Add a new edge to the graph connecting 'n1' and 'n2'; if either n1 or n2 are not lready present in the graph, they should be added."""
+        self.is_node(node1)
+        self.is_node(node2)
+        self.set_of_nodes.add(node1)
+        self.set_of_nodes.add(node2)
+        node1.edges.add(node2)
+
+    def del_edge(self, node1, node2):
+        """Delete the edge connecting 'n1' and 'n2' from the graph, raises a Key Error if no such edge exists."""
+        try:
+            node1.edges.remove(node2)
+        except KeyError:
+            raise KeyError('Edge that you are trying to exist does not exist.')
+        except AttributeError:
+            raise TypeError('{} is not a GNode type.'.format(str(node1)))
+
+    def del_node(self, node_to_del):
+        """Delete a node from the graph, raises an error if no such node exists."""
+        try:
+            self.set_of_nodes.remove(node_to_del)
+        except KeyError:
+            raise KeyError('Node does not exist in graph.')
+
+        for node in self.set_of_nodes:
+            try:
+                node.edges.remove(node_to_del)
+            except KeyError:
+                pass
+
+    def nodes(self):
+        """Return a list of nodes."""
+        return list(self.set_of_nodes)
+
+    def edges(self):
+        """Return a list of edges."""
+        list_of_edges = []
+        for node in self.set_of_nodes:
+            for edge in node.edges:
+                list_of_edges.append(Edge(node, edge))
+        return list_of_edges
+
+    def adjacent(self, node1, node2):
+        """Return true if two nodes are connected by an edge, false if not connected.  Will raise a TypeError is one node is not in the graph."""
+        self.is_node(node1)
+        self.is_node(node2)
+        if not self.has_node(node1) or not self.has_node(node2):
+            raise KeyError('Cannot check adjacent on a node that is not in the graph.')
+        else:
+            return node2 in node1.edges
+
+    def neighbors(self, node):
```

Comment:
How would someone using your data structure get ahold of the node whose neighbors they want to check, in order to pass it into this function?